src/storage/sftp/storage.c

```diff
@@ -831,31 +832,52 @@ storageSftpNew(const String *const path, const String *const host, const unsigne
             THROW_FMT(ServiceError, "libssh2 handshake failed");
 
         int hashType = LIBSSH2_HOSTKEY_HASH_SHA1;
+        size_t hashSize = 0;
 
-        switch (param.hostkeyHash)
+        switch (param.hostkeyHashType)
         {
             case hashTypeMd5:
                 hashType = LIBSSH2_HOSTKEY_HASH_MD5;
+                hashSize = 16;
                 break;
 
             case hashTypeSha1:
                 hashType = LIBSSH2_HOSTKEY_HASH_SHA1;
+                hashSize = 20;
                 break;
 
 #ifdef LIBSSH2_HOSTKEY_HASH_SHA256
             case hashTypeSha256:
                 hashType = LIBSSH2_HOSTKEY_HASH_SHA256;
+                hashSize = 32;
                 break;
 #endif // LIBSSH2_HOSTKEY_HASH_SHA256
 
             default:
-                THROW_FMT(ServiceError, "requested ssh2 hostkey hash type (%s) not available", strZ(strIdToStr(param.hostkeyHash)));
+                THROW_FMT(
+                    ServiceError, "requested ssh2 hostkey hash type (%s) not available", strZ(strIdToStr(param.hostkeyHashType)));
                 break;
         }
 
-        if (libssh2_hostkey_hash(driver->session, hashType) == NULL)
+        const char *binaryFingerprint = libssh2_hostkey_hash(driver->session, hashType);
+
+        if (binaryFingerprint == NULL)
             THROW_FMT(ServiceError, "libssh2 hostkey hash failed: libssh2 errno [%d]", libssh2_session_last_errno(driver->session));
 
+        // Compare fingerprint if provided
+        if (param.hostFingerprint != NULL)
+        {
+            char fingerprint[256];
```

Comment:
Maybe a comment here that 256 is big enough to hold the hex representation of all currently supported hash types (including a mention that we need twice as much space to hold the string-hex version vs. the raw).

Reply:
Addressed in d080b7f9071478ed0730f5ee15efe0b4cfba13b4